model/task/task.go

```diff
@@ -2798,6 +2804,37 @@ func (t *Task) SetResetWhenFinished() error {
 	)
 }
 
+// SetResetWhenFinishedWithInc requests that a task (that was marked to
+// automatically reset when finished via the agent status server) reset itself
+// when finished. It will also increment the number of automatic resets the task
+// has performed.
+func (t *Task) SetResetWhenFinishedWithInc() error {
+	if t.ResetWhenFinished {
+		return nil
+	}
+	err := UpdateOne(
+		bson.M{
+			IdKey:      t.Id,
+			AbortedKey: bson.M{"$ne": true},
```

Comment:
On the same train of thought that a command may fail but may not stop the task immediately, does this still work if a command fails, and the task is aborted later on?

Let's say (same as before) a shell.exec fails in pre but this time the user configured `pre_error_fails_task: true` in their YAML. The task then skips to post. Then later on, the user aborts the task, but the task commands ignore it because aborts don't stop the post commands from running. After the task ends, will it auto-restart because it was originally marked to auto-restart, or will it not because the task was aborted later on? I imagine for reliability, when checking to reset the task, we would have to check if the task was aborted right at the moment when we're deciding whether to reset, rather than here where we're not sure if the task may eventually abort later.

Reply:
You're right that this only prevents the task from being auto restarted if the command with `retry_on_fail` set fails due to an abort rather than failing organically, and if a user sends in a manual abort in between the duration where the command fails and the post-task commands complete, the task will still auto-restart.

I'd say that this would be the behavior we should expect though - if a command with `retry_on_fail` fails organically, that should serve as the agreed criteria for restarting the task, so even if the user sends in a manual abort in the small window where the post commands are running (which doesn't seem like something that would happen too often), the task should still auto-restart.

Reply:
Hmm, I think that it would make more sense to obey the abort though no matter when it happens. It seems reasonable that abort would mean stop and don't auto-restart at all since the user is trying to stop their patch. It would also be complex from the perspective of understanding the feature's behavior, if users see their task have to split their thinking for an aborted task between "did the task abort and cause the auto-restart, or did the auto-restart already get triggered earlier, and then I aborted it later". It may also be a little difficult because they have to abort once, then potentially unschedule/abort again to get the patch to really stop.

Reply:
Added checks such that if a task / display task / single host TG is marked for automatic restart but has _also_ since been aborted, forgo the restart, which should protect against this scenario.